share/revng/test/configuration/revng-qa/runtime-abi-tests.yml

```diff
@@ -0,0 +1,109 @@
+tags:
+  - name: x86-64-systemv-abi
+    implies: [x86-64]
+    variables:
+      ABI_NAME: SystemV_x86_64
+      CFLAGS:
+        - -DABI_SystemV_x86_64
+
+  - name: x86-64-microsoft-abi
+    implies: [x86-64]
+    variables:
+      ABI_NAME: Microsoft_x86_64
+      CFLAGS:
+        - -DABI_Microsoft_x86_64
+        - -maccumulate-outgoing-args
+
+  - name: i386-systemv-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: SystemV_x86
+      CFLAGS:
+        - -DABI_SystemV_x86
+
+  - name: i386-regparm-3-systemv-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: SystemV_x86_regparm_3
+      CFLAGS:
+        - -DABI_SystemV_x86_regparm_3
+
+  - name: i386-regparm-2-systemv-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: SystemV_x86_regparm_2
+      CFLAGS:
+        - -DABI_SystemV_x86_regparm_2
+
+  - name: i386-regparm-1-systemv-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: SystemV_x86_regparm_1
+      CFLAGS:
+        - -DABI_SystemV_x86_regparm_1
+
+  - name: i386-cdecl-microsoft-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: Microsoft_x86_cdecl
+      CFLAGS:
+        - -DABI_Microsoft_x86_cdecl
+
+  - name: i386-fastcall-microsoft-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: Microsoft_x86_fastcall
+      CFLAGS:
+        - -DABI_Microsoft_x86_fastcall
+
+  - name: i386-stdcall-microsoft-abi
+    implies: [i386]
+    variables:
+      ABI_NAME: Microsoft_x86_stdcall
+      CFLAGS:
+        - -DABI_Microsoft_x86_stdcall
+
+  - name: aarch64-aapcs64-abi
+    implies: [aarch64]
+    variables:
+      ABI_NAME: AAPCS64
+      CFLAGS:
+        - -DABI_AAPCS64
+
+  - name: arm-aapcs-abi
+    implies: [arm]
+    variables:
+      ABI_NAME: AAPCS
+      CFLAGS:
+        - -DABI_AAPCS
+
+sources:
+  - tags: [abi-test-function-prototypes, simple-executable]
```

Comment:
@aleclearmind I think I mentioned it somewhere, but just to stress, I'm keeping the `simple-executable` on intact, if that's alright with you.

Reply:
`functions.c` on its own seems to do close to nothing.
Why do we want to build it through the `simple-executable` toolchain?

If there's a reason add a comment, if there's no strong reason, let's drop it.

Reply:
This is the one we use to extract dwarf information from. I'd rather keep it as close as possible to `calc` and friends (binaries we actively test) to limit any possibilities of it being the reason for a failure or something.

Reply:
That just saves from typing `${TRIPLE}gcc $INPUT -o $OUTPUT $CFLAGS`, which basically we already do in `prepare_for_runtime.sh`.
On the other hand, it makes `test-conversion.sh` depend on stuff that's not produced by `prepare_for_runtime.sh`, increasing complexity.

`prepare_for_runtime.sh` produces a directory, `test-conversion.sh` consumes it. That's it.

Reply:
It's less about the current state, and more about "what if the default gcc invocation changes in the future". Also consider that this will also be build with `msvc` too sooner or later.

But, I guess I'm not against pushing the remaining binaries into the same directory.

TL;DR: I kind of just convinced myself (sorry for the noise), let's indeed move them all together.

Reply:
The goal of `revng-test-configure` is to easily build pipelines of commands, repeat them, configure them with tags and the like.

If we need to factor code out, we can use support scripts.
Let's not use the pipeline feature for factoring code, it's not its purpose.

Reply:
Done.